src/JSInterop/Microsoft.JSInterop/src/Infrastructure/DotNetDispatcher.cs

```diff
@@ -292,6 +295,32 @@ public static void EndInvokeJS(JSRuntime jsRuntime, string arguments)
             }
         }
 
+        /// <summary>
+        /// Accepts the byte array data being transferred from JS to DotNet.
+        /// </summary>
+        /// <param name="jsRuntime">The <see cref="JSRuntime"/>.</param>
+        /// <param name="id">Identifier for the byte array being transfered.</param>
+        /// <param name="data">Byte array to be transfered from JS.</param>
+        public static void SupplyByteArray(JSRuntime jsRuntime, long id, byte[] data)
+        {
+            if (id == 0)
+            {
+                // Starting a new transfer, clear out previously stored byte arrays
+                jsRuntime.ResetByteArraysToBeRevived();
+            }
+            else if (id != (jsRuntime.ByteArraysToBeRevived.Count + 1))
+            {
+                throw new ArgumentOutOfRangeException($"Element id '{id}' cannot be added to the byte arrays to be revived with length '{jsRuntime.ByteArraysToBeRevived.Count}'.");
+            }
+            else if ((31 * 1024) - data.Length < jsRuntime.ByteArraysToBeRevivedByteLength) // TODO; get this limit from SignalR somehow?
+            {
+                throw new ArgumentOutOfRangeException("Exceeded the maximum byte array transfer limit for a call.");
+            }
+
+            jsRuntime.ByteArraysToBeRevived.Append(data);
+            jsRuntime.ByteArraysToBeRevivedByteLength += data.Length;
```

Comment:
These two statements should be a single operation. (Append should also update the length)

Reply:
I don't get this suggestion. `ByteArraysToBeRevived` and `ByteArraysToBeRevivedByteLength` are separate things.

Reply:
Append will update the length of `ByteArraysToBeRevived` where length is the number of byte arrays. `ByteArraysToBeRevivedByteLength` tracks the total number of bytes contained in `ByteArraysToBeRevived`. This is done to track whether or not we're within the limits of a single SignalR message.

I've renamed `ByteArraysToBeRevivedByteLength` to `ByteArraysToBeRevivedTotalBytes` to hopefully make this distinction clearer.